http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -505,12 +513,217 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
+    implicit as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- as.ask
+      userToken = org.http4s.headers.Authorization(
+        org.http4s.Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token)
+      )
+      workspaceDescOpt <- wsmDao.getWorkspace(workspaceId, userToken)
+      workspaceDesc <- F.fromOption(workspaceDescOpt, WorkspaceNotFoundException(workspaceId, ctx.traceId))
+
+      hasPermission <- authProvider.hasPermission[WorkspaceResourceSamResourceId, WorkspaceAction](
+        WorkspaceResourceSamResourceId(workspaceId),
+        WorkspaceAction.CreateControlledUserResource,
+        userInfo
+      )
+
+      _ <- F.raiseUnless(hasPermission)(ForbiddenError(userInfo.userEmail))
+
+      appOpt <- KubernetesServiceDbQueries.getActiveFullAppByWorkspaceIdAndAppName(workspaceId, appName).transaction
+      _ <- appOpt.fold(F.unit)(c =>
+        F.raiseError[Unit](AppAlreadyExistsInWorkspaceException(workspaceId, appName, c.app.status, ctx.traceId))
+      )
+
+      samResourceId <- F.delay(AppSamResourceId(UUID.randomUUID().toString))
+      // TODO Verify managed identiy is covered by lookupOriginatingUserEmail
+//      originatingUserEmail <- authProvider.lookupOriginatingUserEmail(userInfo)
+//      _ <- authProvider
+//        .notifyResourceCreated(samResourceId, originatingUserEmail, googleProject)
+//        .handleErrorWith { t =>
+//          log.error(ctx.loggingCtx, t)(
+//            s"Failed to notify the AuthProvider for creation of kubernetes app ${googleProject.value} / ${appName.value}"
+//          ) >> F.raiseError[Unit](t)
+//        }
+
+      cloudContext <- (workspaceDesc.azureContext, workspaceDesc.gcpContext) match {
+        case (Some(azureContext), _) => F.pure[CloudContext](CloudContext.Azure(azureContext))
+        case (_, Some(gcpContext))   => F.pure[CloudContext](CloudContext.Gcp(gcpContext))
+        case (None, None) => F.raiseError[CloudContext](CloudContextNotFoundException(workspaceId, ctx.traceId))
+      }
+
+      saveCluster <- F.fromEither(
+        getSavableCluster(userInfo.userEmail, cloudContext, ctx.now, None, None, Some(workspaceId))
+      )
+      // TODO: Update once AzureCloudContext is populated with AKS information
+      saveClusterResult <- KubernetesServiceDbQueries.saveOrGetClusterForApp(saveCluster).transaction
+      _ <-
+        if (saveClusterResult.minimalCluster.status == KubernetesClusterStatus.Error)
+          F.raiseError[Unit](
+            KubernetesAppCreationException(
+              s"You cannot create an app while a cluster${saveClusterResult.minimalCluster.clusterName.value} is in status ${saveClusterResult.minimalCluster.status}",
+              Some(ctx.traceId)
+            )
+          )
+        else F.unit
+
+      runtimeServiceAccountOpt = none[WorkbenchEmail] // TODO: Figure out how to get PET managed identity
```

Comment:
Yup we'll need to get a pet managed identity for the azure case.. it's a separate Sam call: https://sam.dsde-dev.broadinstitute.org/#/Azure

Reply:
Could you review my updated implementation?

Reply:
Looks good